com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/ZigBeeDongleEzsp.java

```diff
@@ -186,6 +196,11 @@ public class ZigBeeDongleEzsp implements ZigBeeTransportTransmit, ZigBeeTranspor
      */
     private DeviceType deviceType = DeviceType.COORDINATOR;
 
+    /**
+     * The Ember NCP utility class
+     */
+    private EmberNcp ncp;
```

Comment:
I don't think it's a good idea to only allocate a single `EmberNcp` that's used by multiple threads. That saves information in fields there that may be required by users of an individual NCP instance (eg the `getLastStatus()` method). I think this needs to change back to the original implementation so that a new instance of the `EmberNcp` is created for different threads/users.

Reply:
I saw that before I removed the method, I think it's a waste of ressource. The only information saved in the class was the last status (which is used in the library at one place only - energy scan). Client code that would like to access the last status could save it and access it later. Every method in `EmberNcp` returns the response (which includes the status) or the status. In my opinion, this class could be merged into `ZigBeeDongleEzsp` (at the end, every instance of `EmberNcp` was accessed by first getting `ZigBeeDongleEzsp` via the network manager).

I agree it's a breaking change but that could be merged in a next major release. In any case, let me know if you agree with the change. If not, I will revert this part.

Reply:
> Every method in EmberNcp returns the response (which includes the status) or the status.

That's not correct. Many methods return the inner information or null if the response was an error. For example -:

```
    public EmberKeyStruct getKey(EmberKeyType keyType) {
        EzspGetKeyRequest request = new EzspGetKeyRequest();
        request.setKeyType(keyType);
        EzspTransaction transaction = protocolHandler
                .sendEzspTransaction(new EzspSingleResponseTransaction(request, EzspGetKeyResponse.class));
        EzspGetKeyResponse response = (EzspGetKeyResponse) transaction.getResponse();
        if (response == null) {
            return null;
        }
        logger.debug(response.toString());
        lastStatus = response.getStatus();
        if (lastStatus != EmberStatus.EMBER_SUCCESS) {
            return null;
        }
        return response.getKeyStruct();
    }
```

In this case, if you want to know why the key request failed, you need to access `lastStatus`.